packages/dds/tree/src/simple-tree/treeNodeKernel.ts

```diff
@@ -31,9 +34,24 @@ export class TreeNodeKernel implements Listenable<TreeChangeEvents> {
 	public constructor(public readonly node: TreeNode) {}
 
 	public hydrate(anchorNode: AnchorNode): void {
-		const offChildrenChanged = anchorNode.on("childrenChangedAfterBatch", () => {
-			this.#events.emit("nodeChanged");
-		});
+		const offChildrenChanged = anchorNode.on(
+			"childrenChangedAfterBatch",
+			({ changedFields }) => {
+				const flexNode = anchorNode.slots.get(flexTreeSlot);
+				assert(flexNode !== undefined, "Flex node does not exist");
+				const nodeSchema = getSimpleNodeSchema(flexNode.schema);
+				const changedProperties = isObjectNodeSchema(nodeSchema)
+					? new Set(
+							[...changedFields].map(
+								(field) =>
+									nodeSchema.storedKeyToViewKeyMap.get(field) ??
+									fail(`Could not find stored key '${field}' in schema.`),
+							),
+						)
+					: new Set(changedFields);
```

Comment:
You technically shouldn't have to do a defensive copy here since the user recieves the set as Readonly. And, I don't think it matters for our internals anyway since we're done with it at this point.